benchmarks/swe_bench/swe_bench.py

```diff
@@ -0,0 +1,252 @@
+"""SWE-bench: Can Language Models Resolve Real-World GitHub Issues?
+
+Carlos E. Jimenez, John Yang, Alexander Wettig, Shunyu Yao, Kexin Pei, Ofir Press, Karthik Narasimhan
+https://arxiv.org/abs/2310.06770
+"""
+
+import json
+import os
+import re
+import shlex
+from pathlib import Path
+
+from docker import DockerClient
+from swebench.harness.constants import (
+    APPLY_PATCH_FAIL,
+    MAP_REPO_TO_INSTALL,
+    MAP_REPO_VERSION_TO_SPECS,
+    RESET_FAILED,
+    TESTS_ERROR,
+    TESTS_TIMEOUT,
+)
+from swebench.harness.log_parsers import MAP_REPO_TO_PARSER
+from swebench.harness.utils import get_test_directives
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import FieldSpec, Sample, hf_dataset
+from inspect_ai.scorer import Score, Scorer, Target, mean, scorer, std
+from inspect_ai.solver import TaskState, use_tools, system_message, generate
+from inspect_ai.tool import bash, python
+from inspect_ai.util import sandbox
+
+
+INPUT_PROMPT = "Please solve the following issue:\n\n{issue_text}"
+COMPOSE_FILE_DIR = Path(__file__).parent / "resources/compose_files/"
+os.makedirs(COMPOSE_FILE_DIR, exist_ok=True)
+
+SAMPLE_TO_IMAGE_PATH = COMPOSE_FILE_DIR / "sample_to_image.json"
+
+@task
+def swe_bench(
+    instance_id : str,
+    split: str | None = "dev",
+    dataset_name : str ="princeton-nlp/SWE-bench_Lite",
+) -> Task:
+
+    dataset = hf_dataset(
```

Comment:
Maybe factor the synthesis of the `swebench_sample` into a helper function? (we typically do that for less than trivial dataset ingestion)

Reply:
Sure!